### PR TITLE
Establish Room Socket

### PR DESCRIPTION
**Pull request by tausifmuzaffar**
Rooms need to be able to receive and publish bot events using sockets

### CODE REVIEW

File: view/rooms/app.js [hunk 1]
```diff
@@ -75,6 +77,89 @@ function createHeader(bot) {
...
+    'refocus.internal.realtime.bot.action.update';
+  const botDataAdd =
+    'refocus.internal.realtime.bot.data.add';
+  const botDataUpdate =
+    'refocus.internal.realtime.bot.data.update';
+  const botEventAdd =
+    'refocus.internal.realtime.bot.event.add';
+
+  socket.on('connect', () => {
+    console.log('Socket Connected');
```
Comment (iamigo) [inline view/rooms/app.js]: Did you mean to leave console.log on? If you want browser-side debug/logging, please consider putting it behind a flag like we do with the perspectives page (?debug=REALTIME)

File: view/rooms/app.js [hunk 2]
```diff
@@ -75,6 +77,89 @@ function createHeader(bot) {
...
+    Object.keys(botInfo).forEach((key) => {
+      document.getElementById(botInfo[key])
+      .dispatchEvent(new CustomEvent('refocus.events', {
+        detail: events,
+      }));
+    });
+  });
+
+  socket.on('disconnect', () => {
+     console.log('Socket Disconnected');
```
Comment (iamigo) [inline view/rooms/app.js]: same comment as above

File: view/rooms/app.js [hunk 3]
```diff
@@ -105,7 +190,7 @@ function parseBot(bot) {
 
   // go through zipEntries that arent 'index.html'
   const zipEntriesNoIndex = zipEntries.filter(
-    entry => entry.name !== 'index.html'
+    (entry) => entry.name !== 'index.html'
   );
```
Comment (iamigo) [inline view/rooms/app.js]: why not use `zipEntries.filter(...).forEach(...)`?

**Comment by BrandonMurnane** [comment]
@tausifmuzaffar Good to merge if youve given @iamigo his demo (also says im not authorized to merge anyway)

**Comment by tausifmuzaffar** [comment]
@iamigo @kamilsmuga same issue about merging for #698

**Pull request merged**